File: src/test/java/photos/brooklyn/practice/codingexercises/java/leetcode/PrimePalindromTest.java
package photos.brooklyn.practice.codingexercises.java.leetcode;

import org.junit.Test;

import static photos.brooklyn.practice.codingexercises.java.leetcode.PrimePalindrom.*;

import static org.junit.Assert.*;

public class PrimePalindromTest {

    @Test
    public void primePalindromeOdd() {
        assertEquals(new PrimePalindrom().primePalindrome(8),11);
    }

    @Test
    public void primePalindromeEven() {
        assertEquals(new PrimePalindrom().primePalindrome(13),101);
    }

    @Test
    public void primePalindromeTest1() {
        assertEquals(new PrimePalindrom().primePalindrome(9989900),100030001);
    }

    @Test
    public void closestPalindromEvenAlready() {
        assertEquals(closestPalindrom(1221), 1221);
    }

    @Test
    public void closestPalindromEvenSmallBig() {
        assertEquals(closestPalindrom(1231), 1331);
    }

    @Test
    public void closestPalindromEvenBigSmall() {
        assertEquals(closestPalindrom(1431), 1441);
    }

    @Test
    public void isPalindromEven() {
        assertTrue(isPalindrom(1221));
    }

    @Test
    public void isPalindromOdd() {
        assertTrue(isPalindrom(12321));
    }

    @Test
    public void isNotPalindromOdd() {
        assertFalse(isPalindrom(12331));
    }

    @Test
    public void isNotPalindromEven() {
        assertFalse(isPalindrom(2345));
    }

    @Test
    public void nextPalindromEven() {
        assertEquals(nextPalindrom(1221), 1331);
    }

    @Test
    public void nextPalindromOdd() {
        assertEquals(nextPalindrom(12521), 12621);
    }

    @Test
    public void reverseEven() {
        assertEquals(reverse("abcd"), "dcba");
    }
    @Test
    public void reverseOdd() {
        assertEquals(reverse("abc"), "cba");
    }
}

File: src/test/scala/photos/brooklyn/practice/codingexercises/scala/leetcode/PrimePalindromTest.scala
package photos.brooklyn.practice.codingexercises.scala.leetcode

import org.scalatest.{FlatSpec, Matchers}
import photos.brooklyn.practice.codingexercises.scala.leetcode.PrimePalindrom._

class PrimePalindromTest extends FlatSpec with Matchers {

  behavior of "PrimePalindromTest"

  it should "primePalindrome test8" in {
    primePalindrome(8) shouldEqual 11
  }

  it should "primePalindrome test_9989900" in {
    primePalindrome(9989900) shouldEqual 100030001
  }

  it should "primePalindrome test_13" in {
    primePalindrome(13) shouldEqual 101
  }

  it should "primePalindrome test_3" in {
    primePalindrome(3) shouldEqual 3
  }

}
